task/git-clone-oci-ta/0.1/git-clone-oci-ta.yaml

```diff
@@ -252,6 +269,44 @@ spec:
         if [ "${EXIT_CODE}" != 0 ]; then
           exit "${EXIT_CODE}"
         fi
+        if [ "${PARAM_MERGE_TARGET_BRANCH}" = "true" ]; then
+          echo "Merge option enabled. Attempting to merge target branch '${PARAM_TARGET_BRANCH}' into HEAD (${RESULT_SHA})."
+          echo "Fetching target branch '${PARAM_TARGET_BRANCH}'..."
+          git fetch origin "${PARAM_TARGET_BRANCH}"
+          FETCH_EXIT_CODE="$?"
+          if [ "${FETCH_EXIT_CODE}" != "0" ]; then
+            echo "ERROR: Failed to fetch target branch '${PARAM_TARGET_BRANCH}'." >&2
+            exit "${FETCH_EXIT_CODE}"
+          fi
+
+          echo "Merging origin/${PARAM_TARGET_BRANCH} into current HEAD..."
+          git config --global user.email "tekton-git-clone@tekton.dev"
+          git config --global user.name "Tekton Git Clone Task"
+          git merge "origin/${PARAM_TARGET_BRANCH}" --no-commit --no-ff --allow-unrelated-histories
+          MERGE_CHECK_EXIT_CODE="$?"
+
+          if [ "${MERGE_CHECK_EXIT_CODE}" != "0" ]; then
+            echo "ERROR: Merge conflict detected or merge failed before commit." >&2
+            echo "--- Git Status ---"
+            git status
+            echo "------------------"
+            exit "${MERGE_CHECK_EXIT_CODE}"
+          else
+            echo "Merge successful (no conflicts found), committing..."
+            git commit -m "Merge branch '${PARAM_TARGET_BRANCH}' into ${RESULT_SHA}"
+            COMMIT_EXIT_CODE="$?"
+            if [ "${COMMIT_EXIT_CODE}" != "0" ]; then
+              echo "ERROR: Failed to commit merge." >&2
+              exit "${COMMIT_EXIT_CODE}"
+            fi
+            MERGED_SHA=$(git rev-parse HEAD)
+            echo "New HEAD after merge: ${MERGED_SHA}"
+            echo "${MERGED_SHA}" >"$(results.merged_sha.path)"
+          fi
+        else
+          echo "Merge option disabled. Using checked-out revision ${RESULT_SHA} directly."
```

Comment:
If we do not use the merging then the `merged_sha` result will be empty. How might a user want to leverage this parameter? It makes sense that we wouldn't override the `commit` result with it because later tasks/processes might expect this commit to be valid and referenceable.

Reply:
You're correct: when mergeTargetBranch is not enabled, merged_sha will be unset (empty). This is intentional.

We’ve separated commit (which always reflects the SHA of the checked-out revision) from merged_sha (which is only defined if a merge happens). This avoids breaking any downstream consumers that rely on commit to be stable and referenceable.

A user might leverage merged_sha like this:

* Use it in testing pipelines that simulate merged states (similar to how Prow operates with refs/pull/.../merge).
* Have conditional steps that behave differently when a merge occurred (e.g., special test coverage or conflict resolution).
* Use it for tracing/debugging what commit the tests were run on, especially when failures arise from merge behavior.